zeitschrift-fur-kunstgeschichte.csl

```diff
@@ -228,14 +253,29 @@
                 </group>
                 <names variable="editor translator" delimiter=", ">
                   <label form="verb-short" suffix=" "/>
-                  <name/>
+                  <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
                 </names>
               </if>
+              <else-if match="any" variable="translator">
+                <choose>
+                  <if match="none" variable="author">
+                    <names variable="editor">
+                      <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
+                      <label form="short" prefix=" (" suffix=")"/>
+                    </names>
+                  </if>
+                </choose>
+                <group delimiter=" ">
+                  <text variable="title" font-style="italic" suffix=","/>
+                  <text macro="translator"/>
+                  <text macro="collection-with-number"/>
+                </group>
+              </else-if>
```

Comment:
I know that something is unnecessary here but I do not understand the details, you know that. I am not that smart. I am not sure if I can do what you ask.
Anyway my solution works correctly, doesn't it? And zotero doesn't care.
I was just glad that I could fix this somehow. It was not easy for me.